src/main/osd/osd_elements.c

```diff
@@ -578,13 +579,28 @@ static void osdElementEscTemperature(osdElementParms_t *element)
     }
 }
 
+#endif // USE_ESC_SENSOR
+
+#if defined(USE_ESC_SENSOR) || defined(USE_RPM_FILTER)
+int getEscRpm(int i)
+{
+    if (motorConfig()->dev.useDshotTelemetry) {
+        return 100.0f / (motorConfig()->motorPoleCount / 2.0f) * getDshotTelemetry(i);
+    } else {
+        return calcEscRpm(getEscSensorData(i)->rpm);
+    }  
+}
+
 static void osdElementEscRpm(osdElementParms_t *element)
 {
-        if (featureIsEnabled(FEATURE_ESC_SENSOR)) {
-            tfp_sprintf(element->buff, "%5d", osdEscDataCombined == NULL ? 0 : calcEscRpm(osdEscDataCombined->rpm));
-        }
+    for (int i=0; i < getMotorCount(); i++) {
+        char rpmStr[5];
+        itoa(getEscRpm(i), rpmStr, 10);
+        strcat(element->buff, " ");
+        strcat(element->buff, rpmStr);
+    }
 }
```

Comment:
I'm thinking the layout of this element needs to be revised. At the very least the RPM values should be normalized in length (probably 5 characters) otherwise the values in the element are going to be jumping around left/right and will be unreadable.

In general displaying these on a single line is probably not the best practice. The MAX7456 only has 30 characters per row and with 4 motors this element will consume 24 - nearly filling an entire row. Possibly a better layout would be:
```
M1: 12345
M2:  7890
M3: 12345
M4:   990
```

Reply:
I have normalised the length to 5 avoid jumping around left/right.

Reply:
I will make the suggested layout tonight.

Reply:
I have pushed this change. Below is a dvr feed with this change
https://www.youtube.com/watch?v=YzVPsa4ut84